baybe/surrogates/base.py

```diff
@@ -124,60 +134,80 @@ def _make_target_scaler() -> OutcomeTransform | None:
         """Make the output scaler for transforming computational dataframes."""
-        import torch
-
         scaler = self._make_target_scaler()
         if scaler is None:
             return _IDENTITY_TRANSFORM
 
         # TODO: Consider taking into account target boundaries when available
-        scaler(torch.from_numpy(objective.transform(measurements).values))
+        scaler(to_tensor(objective.transform(measurements)))
         scaler.eval()
 
         return scaler
 
-    def transform_inputs(self, df: pd.DataFrame, /) -> pd.DataFrame:
-        """Transform an experimental parameter dataframe."""
-        if self._input_transform is None:
-            raise ModelNotTrainedError("The model must be trained first.")
-        return self._input_transform(df)
+    def posterior(self, candidates: pd.DataFrame, /) -> Posterior:
+        """Compute the posterior for candidates in experimental representation.
+
+        Takes a dataframe of parameter configurations in **experimental representation**
+        and returns the corresponding posterior object. Therefore, the method serves as
+        the user-facing entry point for accessing model predictions.
+        """
+        return self._posterior_comp_rep(
+            to_tensor(self._searchspace.transform(candidates))
+        )
 
-    def transform_outputs(self, df: pd.DataFrame, /) -> pd.DataFrame:
-        """Transform an experimental measurement dataframe."""
-        if self._output_transform is None:
-            raise ModelNotTrainedError("The model must be trained first.")
-        return self._output_transform(df)
+    def _posterior_comp_rep(self, candidates: Tensor, /) -> Posterior:
+        """Compute the posterior for candidates in computational representation.
 
-    def posterior(self, candidates: pd.DataFrame, /) -> Posterior:
-        """Evaluate the surrogate model at the given candidate points."""
-        p = self._posterior(to_tensor(self.transform_inputs(candidates)))
+        Takes a tensor of parameter configurations in **computational representation**
+        and returns the corresponding posterior object. Therefore, the method provides
+        the entry point for queries coming from computational layers, for instance,
+        BoTorch's `optimize_*` functions.
+        """
+        p = self._posterior(self._input_scaler.transform(candidates))
         if self._output_scaler is not _IDENTITY_TRANSFORM:
             p = self._output_scaler.untransform_posterior(p)
         return p
 
     @abstractmethod
     def _posterior(self, candidates: Tensor, /) -> Posterior:
-        """Perform the actual posterior evaluation logic."""
+        """Perform the actual model-specific posterior evaluation logic.
+
+        This method is supposed to be overridden by subclasses to implement their
+        model-specific surrogate architecture. Internally, the method is called by the
+        base class with a **scaled** tensor of candidates in **computational
```

Comment:
In that case, I would propose to use a more descriptive name for `candidates` like `scaled_candidates_cr`. 
In general, it might make sense to add stuff like `_cr` or `_er` at several places to make clear which kind of representation is expected (or explicitly agree to "no suffix means X"), but this is something for another PR.

Reply:
but the equivalent of `measurements->train_x/y` is what for `candidates->?` ?

Reply:
One difference is that for the `fit` methods, we have both `inputs` and `outputs`, which are carried by a single dataframe on the exp_rep layer, but need to come as two separate tensors on the torch layer (since we cannot go via column names there). 

In all our other objects, we use `measurements` to refer to the raw exp_rep data, hence I reused that variable naming in the public `fit` method.

If we now apply the renaming logic from candidates to this situation, we'd get something like `train_x_comp_scaled` and `train_y_comp_scaled`, but there would be no "unscaled" or "exp" versions of it... 🤔 

Does that answer your question?

Reply:
seems were ot talking about the same thing, the candidates are used for inference, hence it doesnt make sense to me to call them anything with `train`

Reply:
Ok, perhaps this entire discussion is not very important right now. In the signatures, I've made the arguments positional-only anyway, which means that we can change the naming any time without destroying someones code. So for the sake of moving on and getting the release out, let me just close the thread. If you disagree, please reopen or ping me in the final merge PR, can still do the renaming there.